codus/codelog
# Add pending changelog for unreleased changes

## Objective
Solves #50. Adds the ability to generate a "pending changelog" of all unreleased changes. Our personal use case for this is so we can easily display a list of changes that should be tested but are not ready to be committed as a "release" yet.

## Notes
Adds two new setting config options:
`default_pending_changelog_filename` which defaults to **CHANGELOG_PENDING.md**
`default_pending_changelog_title` which defaults to **Pending Changes**

I chose **CHANGELOG_PENDING** over **PENDING_CHANGELOG** so that **CHANGELOG.md** and **CHANGELOG_PENDING.md** are right next to each other in the file directory.

I'm not super happy with the naming conventions so I'm definitely open to suggestions about different names than just "pending".

## How to test
Run `rake` for automated testing.
To test manually run `bundle exec bin/codelog pending` while you have unreleased changes then check **CHANGELOG_PENDING.md** for a list of your unreleased changes.

## Breaking changes
No breaking changes.

## Checklist

[CONTRIBUTING]: https://github.com/codus/codelog/blob/master/CONTRIBUTING.md
